packages/cell-toolbar/src/celltoolbartracker.ts

```diff
@@ -322,13 +365,21 @@ export class CellToolbarTracker implements IDisposable {
       return false; // Nothing in the editor
     }
 
-    const codeMirrorLines =
-      editorWidget.node.getElementsByClassName('CodeMirror-line');
+    const codeMirrorLines = editorWidget.node.getElementsByClassName('cm-line');
     if (codeMirrorLines.length < 1) {
       return false; // No lines present
     }
-    const lineRight = codeMirrorLines[0].children[0] // First span under first pre
-      .getBoundingClientRect().right;
+
+    let lineRight = codeMirrorLines[0].getBoundingClientRect().left;
+    const oneChar = document.createElement('span');
+    oneChar.textContent = 'a';
+    codeMirrorLines[0].appendChild(oneChar);
+    const charWidth = oneChar.getBoundingClientRect().width;
+    codeMirrorLines[0].removeChild(oneChar);
+    // Record only the text content's length for each child node
+    codeMirrorLines[0].childNodes.forEach((childNode: ChildNode) => {
+      lineRight += (childNode.textContent?.length || 0) * charWidth;
+    });
```

Comment:
Forced style calculation alert. We can do this without modifying DOM (and get more accurate results at the same time):

```suggestion
    const range = document.createRange();
    range.selectNodeContents(codeMirrorLines[0].childNodes);
    const sizes = range.getClientRects();
    lineRight += [...sizes].reduce((prev, rect) => prev + rect.width, 0);
```

Reply:
Thank you! The proposed line 375 gave me a compilation error for attempting to provide a node list where a single node was expected. I pushed a new change with `setStart` and `setEnd` calls instead.

Reply:
Sorry it should be `range.selectNodeContents(codeMirrorLines[0]);` without `.childNodes`.